### PR TITLE
msa: added support for xacro args

### PR DESCRIPTION
**Pull request by rhaschke**
I stumbled about a robot urdf that required xacro cmdline arguments. This wasn't supported in MSA yet (although there was a TODO as reminder). I added support for specifying xacro arguments, both for the `urdf file` and `package` gui. The xacro args are remembered in `.setup_assistant` file and added to `planning_context.launch`.
Removed hard-coded `--inorder` option, but set `inorder` as default for new projects.

### CODE REVIEW

**Comment by bmagyar** [comment]
Didn't have the opportunity to test the code but the changes all look sound and big thumbs up for cleaning up!

File: moveit_setup_assistant/src/widgets/header_widget.cpp [hunk 1]
```diff
@@ -171,15 +173,15 @@ void LoadPathWidget::btn_file_dialog()
...
+QString LoadPathWidget::getQPath() const
 {
   return path_box_->text();
 }
 
 // ******************************************************************************************
 // Get Std String path
 // ******************************************************************************************
-const std::string LoadPathWidget::getPath()
+std::string LoadPathWidget::getPath() const
```
Comment (davetcoleman) [inline moveit_setup_assistant/src/widgets/header_widget.cpp]: my original return value seems wrong but it should be ``const std::string&`` right?

File: moveit_setup_assistant/src/widgets/header_widget.h [hunk 2]
```diff
@@ -74,27 +74,50 @@ class LoadPathWidget : public QFrame
...
+  std::string getPath() const;
 
   /// Set the path with QString
   void setPath(const QString& path);
 
   /// Set the path with std string
   void setPath(const std::string& path);
 };
+
+/// extend LoadPathWidget with additional line edit for arguments
```
Comment (davetcoleman) [inline moveit_setup_assistant/src/widgets/header_widget.h]: Extend

File: moveit_setup_assistant/src/widgets/header_widget.h [hunk 3]
```diff
@@ -74,27 +74,50 @@ class LoadPathWidget : public QFrame
...
+class LoadPathArgsWidget : public LoadPathWidget
```
Comment (davetcoleman) [inline moveit_setup_assistant/src/widgets/header_widget.h]: why not just modify the original LoadPathWidget function to support args? this seems overly complicated

File: moveit_setup_assistant/src/widgets/start_screen_widget.cpp [hunk 4]
```diff
@@ -306,43 +308,80 @@ void StartScreenWidget::loadFilesClick()
...
+  urdf_file_->setArgsEnabled(rdf_loader::RDFLoader::isXacroFile(path.toStdString()));
+}
 
+bool StartScreenWidget::loadPackageSettings(bool warn)
+{
   // Get the package path
-  if (!createFullPackagePath())
-    return false;  // error occured
+  std::string package_path_input = stack_path_->getPath();
+  // check that input is provided
```
Comment (davetcoleman) [inline moveit_setup_assistant/src/widgets/start_screen_widget.cpp]: capitalize

File: moveit_setup_assistant/src/widgets/start_screen_widget.cpp [hunk 5]
```diff
@@ -306,43 +308,80 @@ void StartScreenWidget::loadFilesClick()
...
+  // set xacro args from loaded settings
+  stack_path_->setArgs(QString::fromStdString(config_data_->xacro_args_));
+}
+
+void StartScreenWidget::onUrdfPathChanged(const QString& path)
+{
+  urdf_file_->setArgsEnabled(rdf_loader::RDFLoader::isXacroFile(path.toStdString()));
+}
 
+bool StartScreenWidget::loadPackageSettings(bool warn)
```
Comment (davetcoleman) [inline moveit_setup_assistant/src/widgets/start_screen_widget.cpp]: warn --> ``show_warnings``

File: moveit_setup_assistant/src/widgets/start_screen_widget.h [hunk 6]
```diff
@@ -55,8 +55,7 @@ namespace moveit_setup_assistant
 {
 // Class Prototypes
 class SelectModeWidget;
-class LoadPathWidget;
-// class LoadURDFWidget;
+class LoadPathArgsWidget;
```
Comment (davetcoleman) [inline moveit_setup_assistant/src/widgets/start_screen_widget.h]: i don't like how you renamed this class, can we keep old name and not have two different classes?

File: moveit_setup_assistant/src/widgets/header_widget.cpp (hunk 1 above)
Comment (rhaschke) [reply moveit_setup_assistant/src/widgets/header_widget.cpp]: Yes, that's another option, which doesn't copy the string.

File: moveit_setup_assistant/src/widgets/header_widget.h (hunk 3 above)
Comment (rhaschke) [reply moveit_setup_assistant/src/widgets/header_widget.h]: I had it this way before, but switch to two classes by intention: There are some usages of LoadPathWidget, that do not require the arguments at all. For me, it was clearer to have separate classes for both use cases. It's not more complicated in my opinion.

File: moveit_setup_assistant/src/widgets/start_screen_widget.h (hunk 6 above)
Comment (rhaschke) [reply moveit_setup_assistant/src/widgets/start_screen_widget.h]: See above. A better name is welcome ;-)

File: moveit_setup_assistant/src/widgets/header_widget.cpp (hunk 1 above)
Comment (rhaschke) [reply moveit_setup_assistant/src/widgets/header_widget.cpp]: No, in this case we actually forward the return value of a function. `const reference` only makes sense when return internal variables.

**Pull request merged**